### PR TITLE
feat(TextArea): Add autoAdjustHeight

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR adds the ability for `TextArea` to auto adjust the height based in the content

cAUtPGDi1P

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -44,9 +45,16 @@ export interface TextAreaProps extends UIComponentProps, ChildrenComponentProps
 
   /** A textarea can take the width of its container. */
   fluid?: boolean;
+
+  /** A textarea can be set to automatically adjust the height */
+  autoAdjustHeight?: boolean;
```
Comment (assuncaocharles) [inline packages/fluentui/react-northstar/src/components/TextArea/TextArea.tsx:50]: I called it `autoAdjustHeight` to be similar to Fabric and it was also as the user named it when requiring it.... But I wonder if `autoResize` wouldn't be a better name

File: packages/fluentui/react-northstar/src/themes/teams/components/TextArea/textAreaStyles.ts [hunk 2]
```diff
@@ -34,6 +35,13 @@ export const textAreaStyles: ComponentSlotStylesPrepared<TextAreaStylesProps, Te
       color: v.disabledColor,
     }),
 
+    ...(p.autoAdjustHeight && {
+      minHeight: v.minHeightAutoAdjust,
+      height: pxToRem(p.autoHeight),
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/themes/teams/components/TextArea/textAreaStyles.ts:40]: Please could you move this `height` to inline style? If I understand correctly - each change in `autoHeight` here would trigger styles change in the document head, which would be more costly performance-wise, than change of the inline style

File: packages/fluentui/react-northstar/src/themes/teams/components/TextArea/textAreaStyles.ts [hunk 3]
```diff
@@ -34,6 +35,13 @@ export const textAreaStyles: ComponentSlotStylesPrepared<TextAreaStylesProps, Te
...
+      overflow: 'hidden',
+      resize: 'vertical',
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/themes/teams/components/TextArea/textAreaStyles.ts:42]: Please would that mean that users could resize the block by themselves? Is that desirable?

File: packages/fluentui/react-northstar/src/themes/teams/components/TextArea/textAreaStyles.ts (hunk 2 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/TextArea/textAreaStyles.ts]: done

File: packages/fluentui/react-northstar/src/components/TextArea/TextArea.tsx [hunk 4]
```diff
@@ -106,21 +126,29 @@ export const TextArea: ComponentWithAs<'textarea', TextAreaProps> &
...
   const handleChange = (e: React.ChangeEvent | React.FormEvent) => {
     const newValue = _.get(e, 'target.value');
-
     _.invoke(props, 'onChange', e, { ...props, value: newValue });
     setValue(newValue);
   };
 
+  React.useLayoutEffect(() => {
+    if (autoAdjustHeight) {
+      inputRef.current.style.height = '0';
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/TextArea/TextArea.tsx:135]:
```suggestion
      inputRef.current?.style.height = '0';
```
?

**Review by pompomon — APPROVED** [review]

File: packages/fluentui/react-northstar/src/components/TextArea/TextArea.tsx (hunk 1 above)
Comment (mnajdova) [reply packages/fluentui/react-northstar/src/components/TextArea/TextArea.tsx:50]: I would vote for `autoResize` it's what I would search for if I need behavior like this.

File: packages/fluentui/react-northstar/src/components/TextArea/TextArea.tsx [hunk 5]
```diff
@@ -106,21 +115,29 @@ export const TextArea: ComponentWithAs<'textarea', TextAreaProps> &
...
     _.invoke(props, 'onChange', e, { ...props, value: newValue });
     setValue(newValue);
   };
 
+  React.useLayoutEffect(() => {
+    if (autoResize) {
+      inputRef.current.style.height = '0';
+      inputRef.current.style.height = `${inputRef.current?.scrollHeight}px`;
+    }
+  }, [inputRef.current?.scrollHeight, autoResize]);
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/TextArea/TextArea.tsx:127]: During the first render `inputRef.current.scrollHeight` will be null. In general it's incorrect to use such things as effect dependencies

File: packages/fluentui/CHANGELOG.md [hunk 6]
```diff
@@ -46,6 +46,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Added icons `FilterIcon`, `MicrophoneDisabledIcon`, `OptionsIcon`, `ReactionsIcon` @TanelVari ([#14745](https://github.com/microsoft/fluentui/pull/14745))
 - Add `foreground5` and `foreground6` in the colorScheme's default definition @assuncaocharles ([#14818](https://github.com/microsoft/fluentui/pull/14818))
 - Add `Form.Datepicker` component @pompomon ([#14820](https://github.com/microsoft/fluentui/pull/14820))
+- Add `autoAdjustHeight` to `TextArea` component @assuncaocharles ([#14860](https://github.com/microsoft/fluentui/pull/14860))
```
Comment (layershifter) [inline packages/fluentui/CHANGELOG.md:49]: PR title and changelog entry have wrong prop name

**Comment by layershifter** [comment]
https://codesandbox.io/s/microsoftfluentui-codesandbox-react-northstar-template-forked-czevk?file=/src/index.tsx

### `border` & `minHeight`

The combination of `border` and `minHeight` breaks the logic:
textarea-bug

---

You can also go through linked issues in https://github.com/Semantic-Org/Semantic-UI-React/pull/3465, there are numerous interesting issues 🔫 I suggest to revert this an use [`react-textarea-autosize`](https://www.npmjs.com/package/react-textarea-autosize) instead.

**Pull request merged**